### PR TITLE
ISLANDORA-2342

### PR DESCRIPTION
**Pull request by jonathangreen**
## JIRA Ticket

https://jira.duraspace.org/browse/ISLANDORA-2342

## What does this Pull Request do?

Update hook 7106, in specific this line https://github.com/Islandora/islandora_xml_forms/blob/7.x/builder/xml_form_builder.install#L525 fails if there is an actual XML association with a NULL value for XSLT transform value.

## What's new?

Fix issue with previous update hook where we set the `transforms` column to not null that may contain null values. Instead of making this column not null we just allow this column to contains nulls.

This commit does three things:
  - Updates the schema to allow nulls in the `transforms` column
  - Fixes the bug in the update hook, so users shouldn't see the issue
  - Adds a new update hook to make sure the column is the same for everyone.

## How should this be tested?

I believe there are 3 scenarios that need testing here: 
1. The user hasn't updated and they have NULLs in the `transforms` column. 
2. A new install of the module, where the new schema is installed automatically. 
3. An install where there are no NULL in the `transforms` column and 7106 has applied successfully. 7107 should apply and allow the transforms column to contains NULLs. 

In each of these scenarios, need to make sure that the form association creations are working well after this patch is applied.

# Interested parties
@Islandora/7-x-1-x-committers @DiegoPino

### CODE REVIEW

**Comment by jonathangreen** [comment]
Release pull here: 
https://github.com/Islandora/islandora_xml_forms/pull/260

**Comment by DiegoPino** [comment]
@jonathangreen++ great!!. Will wait for travis-CI to go green and will approve both. @bryjbrown @bondjimbond this fixes the issue so once both get merged blocker is gone for good.

**Review by DiegoPino — APPROVED** [review]
Well done

**Comment by DiegoPino** [comment]
@jonathangreen @bondjimbond @bryjbrown I omitted the 24hours waiting period since we are about to go release and there was agreement this was a blocker, i hope i did well.

**Comment by bryjbrown** [comment]
@DiegoPino @jonathangreen BEAUTIFUL :stars:

**Pull request merged**
